### PR TITLE
OCPCLOUD-2061: Rebase dependencies to k8s 1.27

### PR DESCRIPTION
**Pull request by elmiko**
also update the controller-runtime dependency and fix some issues related to the changes.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/lgtm cancel

Panics in the logs, https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-autoscaler-operator/274/pull-ci-openshift-cluster-autoscaler-operator-master-e2e-aws-operator/1663865391102824448/artifacts/e2e-aws-operator/gather-extra/artifacts/pods/openshift-machine-api_cluster-autoscaler-operator-647759f49d-vhbh2_cluster-autoscaler-operator.log

**Comment by elmiko** [comment]
looking at the output, my best guess is that this is related to the controller-runtime changes.

**Comment by elmiko** [comment]
> Looks like the panic is coming because the logger is nil, can we set up a logger easily?

i hope so

**Comment by JoelSpeed** [comment]
[InjectDecoder](https://github.com/openshift/cluster-autoscaler-operator/blob/f7f168cc5145da8b8c781122a39582e1a2ad39ce/pkg/controller/clusterautoscaler/validator.go#L205-L209) is no longer being called as of v0.15.0 of controller runtime (the client one probably isn't either?) so we need to update the constructor of the validator to avoid using these inject style methods, the code is panicking in the last instance because the decoder is nil

**Comment by elmiko** [comment]
@JoelSpeed ptal, it seems like we are back to normal autoscaler failures, i think CAO is happy now.

File: pkg/controller/clusterautoscaler/validator.go [hunk 1]
```diff
@@ -27,8 +28,10 @@ type Validator struct {
 
 // NewValidator returns a new Validator configured with the given
 // ClusterAutoscaler singleton resource name.
-func NewValidator(name string) *Validator {
+func NewValidator(name string, client client.Client, scheme *runtime.Scheme) *Validator {
```
Comment (JoelSpeed) [inline pkg/controller/clusterautoscaler/validator.go:31]: Should we drop the `InjectDecoder` and `InjectClient` functions now then?

File: pkg/controller/machineautoscaler/validator.go [hunk 2]
```diff
@@ -20,8 +21,11 @@ type Validator struct {
 }
 
 // NewValidator returns a new Validator.
-func NewValidator() *Validator {
-	return &Validator{}
+func NewValidator(client client.Client, scheme *runtime.Scheme) *Validator {
```
Comment (JoelSpeed) [inline pkg/controller/machineautoscaler/validator.go:24]: Same again, should we drop the inject functions?

**Comment by dtobolik** [comment]
Regression looks good on this one, ping me for any major changes
/label qe-approved

**Pull request merged**